README.md

```diff
@@ -811,6 +813,23 @@ RuboCop::RakeTask.new(:rubocop) do |task|
 end
 ```
 
+## Caching
+
+Large projects containing hundreds or even thousands of files can take
+a really long time to inspect, but RuboCop has functionality to
+mitigate this problem. A result caching mechanism is activated if the
+configuration parameter `AllCops: UseCache` is `true`, which it is by
+default. The command line option `--cache false` can be used to turn
+off result caching, thus overriding the configuration parameter. If
+`AllCops: UseCache` is set to `false` in the local `.rubocop.yml`,
+then it's `--cache true` that overrides the setting.
+
+The cache is stored in `.rubocop_cache/` under the current directory.
+There are subdirectories for each user, so as to minimize the risk of
+collisions and file curruption from simultaneous `rubocop`
+invocations. Git users should add `.rubocop_cache` to their
```

Comment:
I don't think we've factored in simultaneous RuboCop runs so far. I really don't want to have the cache stored in the actually directories where rubocop is invoked. People should also keep in mind that while normally people invoke it on a single directory or file there's nothing stopping you from invoking it on a ton of unrelated folders. Guess a simple solution would be to simply monitor the target directory structure under `/temp` or something like this and to create some locks while updating the cache.